src/graphql/resolvers/ssb.js

```diff
@@ -219,62 +202,59 @@ module.exports = function Resolvers () {
             })
             .catch(err => cb(err))
         }, 5),
-        pull.collect((err, messages) => {
-          if (err) reject(err)
-          else resolve(messages)
-        })
+        pull.collect((err, res) => err ? reject(err) : resolve(res))
       )
     })
   }
 
   return {
-    ssb,
-    resolvers: {
-      Query: {
-        getProfile: (_, opts) => getProfile(opts.id),
-        getProfiles: (_, opts) => getProfiles(opts)
-      },
+    Query: {
+      getProfile: (_, opts) => getProfile(opts.id),
+      getProfiles: (_, opts) => getProfiles(opts)
+    },
 
-      Profile: {
-        image: (parent) => toSSBUri(parent.image, { port: ssbPort }),
-        threads: (parent, opts) => getThreads(parent.id, opts),
-        followers: async (parent) => {
-          const ids = await getFollowersIds(parent.id)
-          return getProfilesForIds(ids)
-        },
-        followersCount: async (parent) => {
-          const ids = await getFollowersIds(parent.id)
-          return ids?.length
-        },
-        following: async (parent) => {
-          const ids = await getFollowingIds(parent.id)
-          return getProfilesForIds(ids)
-        },
-        followingCount: async (parent) => {
-          const ids = await getFollowingIds(parent.id)
-          return ids?.length
-        }
+    Profile: {
+      image: (parent) => {
+        if (!parent.image) return
+        return toSSBUri(parent.image, { port: ssbPort })
       },
-
-      Thread: {
-        messages: (parent) => mapMessages(parent.messages)
+      threads: (parent, opts) => getThreads(parent.id, opts),
+      followers: async (parent) => {
+        const ids = await getFollowersIds(parent.id)
+        return getProfilesForIds(ids)
```

Comment:
I think these should be rolled together - why do one streams, await it to finish, then do another steam?
Unless getFollowerIds is used somewhere else I would recommend just having `getFollowerProfiles`.

Reply:
oh I see... you have a count thing as well which doesn't need to load the profiles...
Ok yeah I get this, it's fine. Lets just watch timing of all requests